src/System/reachability.rs

```diff
@@ -0,0 +1,198 @@
+use edbm::zones::OwnedFederation;
+
+use crate::ModelObjects::component::{State, Transition};
+use crate::TransitionSystems::{LocationID, TransitionSystem};
+use std::collections::HashMap;
+use std::rc::Rc;
+
+/// This hold the result of a reachability query
+pub struct Path {
+    pub path: Option<Vec<Transition>>,
+    pub was_reachable: bool,
+}
+
+struct SubPath {
+    previous_sub_path: Option<Rc<SubPath>>,
+    destination_state: State,
+    transition: Option<Transition>,
+}
+
+fn is_trivially_unreachable(end_state: &State, _system: &dyn TransitionSystem) -> bool {
+    if let Some(invariants) = end_state.get_location().get_invariants() {
+        if !&end_state.zone_ref().has_intersection(invariants) {
+            return true;
```

Comment:
pls no returnerino

Reply:
I would say making an else case here to return implicitly will look worse, and will make it harder to expand this section.

Reply:
Not possible,  gets "Mismatched types expected '()', found bool